etcdserver/server.go

```diff
@@ -836,8 +836,14 @@ func (s *EtcdServer) snapshot(snapi uint64, confState raftpb.ConfState) {
 		if err := s.r.storage.SaveSnap(snap); err != nil {
 			log.Fatalf("etcdserver: save snapshot error: %v", err)
 		}
+		log.Printf("etcdserver: saved snapshot at index %d", snap.Metadata.Index)
 
-		err = s.r.raftStorage.Compact(snapi)
+		// keep some in memory entries for slow followers.
+		compacti := uint64(1)
```

Comment:
i think the code is clear enough. and we do not comment code path in general. 
you need to check the compact comment to see what is going on.

```
// Compact discards all log entries prior to i.
// It is the application's responsibility to not attempt to compact an index
// greater than raftLog.applied.
```